src/software/thunderscope/robot_diagnostics/controller_diagnostics.py

```diff
@@ -0,0 +1,287 @@
+
+    def __event_loop(self):
+        # TODO (#3165): Use trace level logging here
+        # logging.debug("Starting handheld controller event handling loop")
+        try:
+            for event in self.controller.read_loop():
+                if self.__stop_thread_signal_event.isSet():
+                    return
+                else:
+                    self.__process_event(event)
+
+        except OSError as ose:
+            logging.debug(
+                "Caught an OSError while reading handheld controller event loop!"
+            )
+            logging.debug("Error message: " + str(ose))
+            logging.debug("Check physical handheld controller USB connection")
+            return
+        except Exception as e:
+            logging.critical(
+                "Caught an unexpected error while reading handheld controller event loop!"
+            )
+            logging.critical("Error message: " + str(e))
+            return
+
+    def __process_move_event_value(self, event_type, event_value) -> None:
+        if event_type == "ABS_X":
+            self.motor_control.direct_velocity_control.velocity.x_component_meters = self.__parse_move_event_value(
+                MoveEventType.LINEAR, event_value
+            )
+
+        elif event_type == "ABS_Y":
+            self.motor_control.direct_velocity_control.velocity.y_component_meters = self.__parse_move_event_value(
+                MoveEventType.LINEAR, event_value
+            )
+
+        elif event_type == "ABS_RX":
+            self.motor_control.direct_velocity_control.angular_velocity.radians_per_second = self.__parse_move_event_value(
+                MoveEventType.ROTATIONAL, event_value
+            )
+
+    def __process_event(self, event):
+        kick_power = 0.0
+        dribbler_speed = 0.0
+
+        abs_event = categorize(event)
+        event_type = ecodes.bytype[abs_event.event.type][abs_event.event.code]
+
+        # TODO (#3165): Use trace level logging here
+        # logging.debug(
+        #     "Processing controller event with type "
+        #     + str(event_type)
+        #     + " and with value "
+        #     + str(abs_event.event.value)
+        # )
+
+        # TODO (#3175): new python versions have a pattern matching,
+        #  that could be useful for handling lots of if statements.
+        #  Some documentation for future:
+        #  https://peps.python.org/pep-0636/
+        #  https://docs.python.org/3/reference/compound_stmts.html#match
+        if event.type == ecodes.EV_ABS:
+            if event_type in ["ABS_X", "ABS_Y", "ABS_RX"]:
+                self.__process_move_event_value(event_type, abs_event.event.value)
+
+        if event_type == "ABS_HAT0X":
+            dribbler_speed = self.__parse_kick_event_value(abs_event.event.value)
+
+        if event_type == "ABS_HAT0Y":
+            kick_power = self.__parse_dribble_event_value(abs_event.event.value)
+
+        if event_type == "ABS_RZ" or "ABS_Z":
```

Comment:
This will always be true since strings are truthy
```suggestion
        if event_type == "ABS_RZ" or event_type == "ABS_Z":
```

Reply:
good catch